internal/dao/pgx/account_pgx.go

```diff
@@ -72,7 +74,17 @@ func (x *accountDao) GetOrCreateByIdentity(ctx context.Context, orgId string, ac
 		LIMIT 1;`
 
 	err := pgxscan.Get(ctx, db.Pool, result, query, orgId, account)
-	if err != nil {
+
+	if errors.Is(err, dao.ErrNoRows) {
+		// Step 4: requery in case transaction isolation (simultaneous transactions)
+		if result.ID == 0 && result.OrgID == "" {
+			zerolog.Ctx(ctx).Warn().Bool("requery", true).Msgf("Organization id %s requery", orgId)
+			err := pgxscan.Get(ctx, db.Pool, result, query, orgId, account)
+			if err != nil {
+				return nil, fmt.Errorf("pgx requery error: %w", err)
+			}
+		}
+	} else if err != nil {
```

Comment:
Linter doesn't like shadowing the var.
We can use the same var and use the same if for catching all the "leftover" errors in the main, right?

Smthing like:
```suggestion
			err = pgxscan.Get(ctx, db.Pool, result, query, orgId, account)
		}
	}
	if err != nil {
```


or just use `requeryErr` var would also work :)

Reply:
Rebased